### PR TITLE
972-topology-card-context-menu-position-fixed-wont-hidden-less-height-accommodate

### LINKED ISSUE #942: The topology card action bar is causing overflow on topology parent container.
Ideally, this should be the top most element over everything else.

This is more pronounced in the incident details page but to some extent also available in topology page.

image

### LINKED ISSUE #972: Topology card context menu position should be fixed so that it won't be hidden when there is less height to accommodate it.

### PR DESCRIPTION
**Pull request by mainawycliffe**
Prevent overflow showing when there isn't enough height

Closes #972 and Closes #942

### CODE REVIEW

**Review by Prashanth-Nelli — CHANGES_REQUESTED** [review]

File: src/components/TopologyCard/TopologyDropdownMenu.tsx [hunk 1]
```diff
@@ -31,6 +31,17 @@ interface IProps {
 }
 
 export const TopologyDropdownMenu = ({ topology, onRefresh }: IProps) => {
+  const dropdownMenuRef = useRef<HTMLDivElement>(null);
+
+  const verticalIconClasses = useMemo(() => {
+    const right = dropdownMenuRef.current?.getBoundingClientRect().right;
+    const top = dropdownMenuRef.current?.getBoundingClientRect().top;
+    if (!right || !top) {
+      return `right-[${right}] top-[${top}]`;
```
[the remark this answers is not in the document]
Comment (mainawycliffe) [reply src/components/TopologyCard/TopologyDropdownMenu.tsx:40]: Nice catch; for some reason, it's still working wondering why now?

[the remark this answers is not in the document]
Comment (Prashanth-Nelli) [reply src/components/TopologyCard/TopologyDropdownMenu.tsx:40]: we can remove this block if it is not required.

[the remark this answers is not in the document]
Comment (mainawycliffe) [reply src/components/TopologyCard/TopologyDropdownMenu.tsx]: This is fixed now

**Review by Prashanth-Nelli — APPROVED** [review]
LGTM

**Pull request merged**